controller/pagecontroller.php

```diff
@@ -40,7 +40,18 @@ public function __construct($appName, IRequest $request, $userId, $cacheManager,
 	 */
 	public function index() {
 		$params = array('user' => $this -> userId,'devices'=>$this->locationManager->loadAll($this->userId));
-		return new TemplateResponse('maps', 'main', $params);
+		$response = new TemplateResponse('maps', 'main', $params);
+		if (class_exists('OCP\AppFramework\Http\ContentSecurityPolicy')) {
+			$csp = new \OCP\AppFramework\Http\ContentSecurityPolicy();
+			// map tiles
+			$csp->addAllowedImageDomain('http://*.mqcdn.com');
+			// marker icons
+			$csp->addAllowedImageDomain('https://api.tiles.mapbox.com');
+			// images inside scripts
```

Comment:
I don't completely understand this yet...
@LukasReschke Is `$csp->addAllowedScriptDomain('\'self\'');` the reason `<script>alert(1)</script>` is possible?
Because I believe this is enabled by default either way (see [contentsecuritypolicy.php](https://github.com/owncloud/core/blob/a9d1a0144018e60ba2728708bf965b4d9855920b/lib/public/appframework/http/contentsecuritypolicy.php#L36))